plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/JavaToJKTreeBuilder.kt

```diff
@@ -960,13 +967,27 @@ class JavaToJKTreeBuilder(
         }
 
         fun PsiLocalVariable.toJK(): JKLocalVariable {
+            val codeBlock = parentOfType<PsiCodeBlock>()
```

Comment:
This way you will calculate mutability of all fields in the block for each field which may result in quadratic complexity. Let's make the mutability computation to be performed only once per each block